data/hai/hai reveal 1 intro.txt

```diff
@@ -827,23 +829,24 @@ mission "Hai Leaks Response 4A"
 		personality staying marked mute
 		system "Waypoint"
 		ship "Geodesic Geocoris (Phylactery)" "Phylactery"
-		dialog `After successfully boarding the Geocoris and defeating its pirate crews, you are greeted by the relieved captive Hai. You promise to escort their ship safely to Allhome. (You will need to swap the Geocoris to your flagship, and depart and land again in order to progress the missions properly.)`
+		dialog `After successfully boarding the Geocoris and defeating its pirate crews, you are greeted by the relieved captive Hai. You send crew aboard to fly the ship and promise to return them safely to Allhome.`
 #	on fail
 #		dialog `You have failed to recover the Geocoris. This is not critical, but does have long-term consequences. If you want the most complete ending, revert to the autosave or another earlier snapshot of the game.`
 #Too hand-holdy for present implementation and content.
 
 
 # Must be able to resolve earlier than the rest in this set.
-#The Geodesic Geocoris may become available to purchase specially later on.
+# The Geodesic Geocoris may become available to purchase specially later on.
 mission "00 Hai Leaks Response 4B"
 	landing
 	source "Allhome"
+	to offer
+		has "outfit (installed): Failed Control System"
```

Comment:
The outfit existed before because that's the only thing we could check, but now we're able to check what model of ships the player has, removing the need for the dummy outfit.
```suggestion
		has "ship model (all): Geodesic Geocoris"
```

Reply:
While this is true, if people are using plugins like Omnis etc. then they could theoretically have a ship of this model already. 
Using the outfit which is only on the variant used here is more secure as a mission functionality.

I'm also keen (at some point in a future plot) to make GG's purchasable, and to use the need to replace the control system to do so as a plot device if you still possess the ship then.

Reply:
Except if people were using plugins like Omnis, they could also get this outfit. We should not be creating missions based on what players might be able to do with plugins.

And if you intend to only make the ship purchasable after a later story which requires you to still have the ship present, you could similarly check on ownership of the ship instead of the outfit for that mission as well, since at the time it will still be the only Geodesic Geocoris.

Reply:
The presence of the outfit is, though, effectively a plot hook to encourage people to look for such a story. 
It's also thematic, in that the ship isn't a refined product, and it's evidence as a reminder of that.

Also things like Omnis don't necessarily add every single possible outfit, while it does add all the ships.

Reply:
I agree with Grey, best to use the outfit instead of the ship since now its linked to the ship (and is like doing a special variant of the ship, which we cant check for without doing a duplicate ship definition)